### DISCUSSION TITLE
Database is not properly configured.

### DISCUSSION THREAD

**Discussion by iamshulz** [jenssegers/laravel-mongodb#2537] [Q&A]
I have properly set all the configurations for Laravel .env and database.php
Also ran db with `mongod` at the terminal w/ admin privilege.
What am I missing here?

.env
```
DB_CONNECTION=mongodb
DB_HOST=127.0.0.1
DB_PORT=27017
DB_DATABASE=homestead
DB_USERNAME=
DB_PASSWORD=
```

database.php
```
'mongodb' => [
            'driver' => 'mongodb',
            'host' => env('DB_HOST', 'localhost'),
            'port' => env('DB_PORT', 27017),
            'database' => env('DB_DATABASE', 'homestead'),
            'username' => env('DB_USERNAME'),
            'password' => env('DB_PASSWORD'),
            'options' => [],
        ],
```

![image](https://github.com/jenssegers/laravel-mongodb/assets/20100579/0df48d07-5c4a-4855-acc7-9b05ef6f61a4)
![image](https://github.com/jenssegers/laravel-mongodb/assets/20100579/975be64b-0de4-4ccb-adbd-784e41a83221)

How is my database not properly configured?

I'm sure that this is more on db issues since it indicates `not properly configured` rather than `not configured`

**Comment by zigzagdev** [comment]
@jamols09 

Hi

### Short answer

Your MongoDB is **only bound to `127.0.0.1`**, but your Laravel application is **running in a different network namespace (most likely a Docker container)**.  
As a result, `127.0.0.1` from Laravel does **not** point to the MongoDB process.

---

### Explanation

Your MongoDB log shows:

```text
Listening on 127.0.0.1
Waiting for connections on port 27017
```

This means MongoDB accepts connections only from the same host/network namespace.

If Laravel is running inside Docker (or another VM), `127.0.0.1` refers to the container itself, not the host machine where MongoDB is running, so the connection fails.


### How to fix it
- Run MongoDB as a Docker service and connect via the service name:
```php
DB_CONNECTION=mongodb
DB_HOST=mongodb
DB_PORT=27017
DB_DATABASE=homestead
```

`mongodb` is the service name defined in docker-compose.yml


Hope this helps !! 🙌
